snap/snapcraft.yaml

```diff
@@ -0,0 +1,108 @@
+name: mysql-server # you probably want to 'snapcraft register <name>'
+base: core22 # the base snap is the execution environment for this snap
+version: '8.0' # just for humans, typically '1.2+git' or '1.3.2'
+summary: MySQL server in a snap. # 79 char long summary
+description: |
+  The MySQL software delivers a very fast, multithreaded, multi-user, 
+  and robust SQL (Structured Query Language) database server. MySQL 
+  Server is intended for mission-critical, heavy-load production 
+  systems as well as for embedding into mass-deployed software.
+
+grade: stable # must be 'stable' to release into candidate/stable channels
+confinement: strict # use 'strict' once you have the right plugs and slots
+
+system-usernames:
+  snap_daemon: shared
+
+layout:
+  /var/lib/mysql-files: 
+    bind: $SNAP_DATA/var/lib/mysql-files
+  /var/lib/mysql:
+    bind: $SNAP_DATA/var/lib/mysql
+  /var/log/mysql:
+    bind: $SNAP_DATA/var/log/mysql
+  /etc/mysql:
+    bind: $SNAP_COMMON/mysql
+  /var/mysqld:
+    bind: $SNAP_DATA/var/run/mysqld
+  /usr/lib/mysql-router:
+    symlink: $SNAP/usr/lib/mysql-router
+  /etc/mysqlrouter:
+    symlink: $SNAP_COMMON
+
+apps:
+  mysql:
+    command: usr/bin/mysql
+  mysql-config-editor:
+    command: usr/bin/mysql_config_editor
+  mysqldump:
+    command: usr/bin/mysqldump
+  mysqlpump:
+    command: usr/bin/mysqlpump
+  mysqlslap:
+    command: usr/bin/mysqlslap
+  mysqladmin:
+    command: usr/bin/mysqladmin
+  mysqld:
+    command: start-mysqld.sh
+    daemon: simple
+    install-mode: disable
+    plugs:
+      - network
+      - network-bind
+  mysqldumpslow:
+    command: usr/bin/mysqldumpslow
+  mysqlrepair:
+    command: usr/bin/mysqlrepair
+  mysql-ssl-rsa-setup:
+    command: usr/bin/mysql_ssl_rsa_setup
+  mysqlanalyze:
+    command: usr/bin/mysqlanalyze
+  mysqlimport:
+    command: usr/bin/mysqlimport
+  mysqlreport:
+    command: usr/bin/mysqlreport
+  mysql-tzinfo-to-sql:
+    command: usr/bin/mysql_tzinfo_to_sql
+  mysqlbinlog:
+    command: usr/bin/mysqlbinlog
+  mysql-migrate-keyring:
+    command: usr/bin/mysql_migrate_keyring
+  mysql-secure-installation:
+    command: usr/bin/mysql_secure_installation
+  mysql-upgrade:
+    command: usr/bin/mysql_upgrade
+  mysqlcheck:
+    command: usr/bin/mysqlcheck
+  mysqloptimize:
+    command: usr/bin/mysqloptimize
+  mysqlshow:
+    command: usr/bin/mysqlshow
+  mysqlrouter:
+    command: start_mysql_router.sh 
+  mysqlrouter-service:
+    command: start_mysql_router.sh
+    daemon: simple
+    install-mode: disable
+    plugs:
+      - network
+      - network-bind
+  mysqlrouter-passwd:
+    command: usr/bin/mysqlrouter_passwd
+  mysqlrouter-keyring:
+    command: usr/bin/mysqlrouter_keyring
+  mysqlrouter-plugin-info:
+    command: usr/bin/mysqlrouter_plugin_info
+
+parts:
+  mysql-server-deb:
+    plugin: nil
+    stage-packages:
+      - mysql-server-8.0
+      - mysql-router
+      - util-linux
+    stage-snaps:
+      - xtrabackup
```

Comment:
since we are now creating a `charmed-mysql-snap`, should we be doing something (perhaps downloading from source) to ensure that versions are aligned?